### PR TITLE
Add Javadoc to Pair, Twin, Tuples, PrimitiveTuples, ObjectPrimitivePa…

### LINKED ISSUE #390: Update Interface Javadocs to suggest possible ways of initialization.
I'm having trouble coming with a short description, but let me tell you my hardship.

I like that I can have all this new entity objects like IntObjectPair. But everything is an interface, and it's not easy to find where and what is the correct factory (method) for an instance. In this case, I had to see implementations of IntObjectPair (IntObjectPairImpl) and search for usages of IntObjectPairImpl's constructor to find out that the factory method is PrimitiveTuples.pair().

So, this information could be in the Javadoc of the interface. IntObjectPair javadoc only has "@see Pair", and Pair only has trivial information. A simple comment like "To get instances of this interface, use PrimitiveTuples.pair()" or even a "@see PrimitiveTuples.pair()" would help tremendously.

Thanks!
Leo.

### PR DESCRIPTION
**Pull request by donraab**
…ir, PrimitivePrimitivePair and PrimitiveObjectPair. Fixes #390.

Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

**Comment by leomrlima** [comment]
If that's automatically created, how do you guarantee that the comments will not vanish on next change?

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/tuple/primitive/PrimitiveTuples.java [hunk 1]
```diff
@@ -91,6 +91,9 @@
 import org.eclipse.collections.api.tuple.primitive.ShortObjectPair;
 import org.eclipse.collections.api.tuple.primitive.ShortShortPair;
 
+/**
+ * This class should be used to create instances of object/primitive, primitive/primitive and primitive/primitive pairs.
```
Comment (leomrlima) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/tuple/primitive/PrimitiveTuples.java]: This looks wrong: last two are the same, maybe it was supposed to be primitive/object?

Comment (donraab) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/tuple/primitive/PrimitiveTuples.java]: good catch.  fixed.

**Comment by donraab** [comment]
I changed the comments in the templates, so they get generated with the code.

**Comment by leomrlima** [comment]
I see. So why doesn't it show commit changes in the generated files as well?

**Comment by donraab** [comment]
The generated files are not checked in.

**Comment by leomrlima** [comment]
Who/what generates them, then?

**Comment by donraab** [comment]
The build process generates the primitive code using StringTemplate.

**Pull request merged**